squirrel/orders/tests/testView.py

```diff
@@ -172,116 +173,178 @@ def test_require_view_permissions_ok(self):
+
+    def test_non_privileged_can_not_change(self):
+        order = Order.objects.create(product=self.product, team=self.team_a)
+        self.client.login(username="loc_engel", password="loc_engel")
+        response = self.post_order(order.id, "REQ")
         self.assertEqual(response.status_code, 403)
 
-    def test_require_change_permission_ok(self):
-        order = Order.objects.create(product=self.product, team=self.team)
-        order.save()
-        self.client.login(username="order_admin", password="order_admin")
-        response = self.client.post(
-            "/orders/{}".format(order.id),
-            {
-                "amount": 1,
-                "product": self.product.id,
-                "team": self.team.id,
-                "unit_price": 1,
-                "state": "REQ",
-            },
-        )
+    def test_change_permission_can_change_anything(self):
+        order = Order.objects.create(product=self.product, team=self.team_a)
+        self.view_user.user_permissions.add(self.change_permission)
+        self.client.login(username="loc_engel", password="loc_engel")
+        response = self.post_order(order.id, "REQ")
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, "/orders")
+        response = self.post_order(order.id, "REQ", 2)
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, "/orders")
+        response = self.post_order(order.id, "REQ", 2, 17.00)
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, "/orders")
+        response = self.post_order(order.id, "APP")
         self.assertEqual(response.status_code, 302)
         self.assertEqual(response.url, "/orders")
+        response = self.post_order(order.id, "REA")
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, "/orders")
+        response = self.post_order(order.id, "COM")
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, "/orders")
+
+    def test_team_members_can_change_some_fields(self):
+        self.team_a.members.add(self.user)
+        my_order = Order.objects.create(product=self.product, team=self.team_a)
+        order = Order.objects.create(product=self.product, team=self.team_b)
+        self.client.login(username="engel", password="engel")
+        # we can not change the state
+        response = self.post_order(my_order.id, "APP")
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, "APP is not one of the available choices")
+        # we can change the amount
+        response = self.post_order(my_order.id, "REQ", 2)
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, "/orders")
+        # we can not change the unit_price
+        # response = self.post_order(my_order.id,"REQ",2,17.00)
```

Comment:
Why is this commented?

Reply:
because users can change the unit price if in state REQuested. 
Users can change almost everything until the order is APProved.